sdks/router-sdk/src/entities/mixedRoute/trade.ts

```diff
@@ -193,19 +196,21 @@ export class MixedRouteTrade<TInput extends Currency, TOutput extends Currency,
     amount: TTradeType extends TradeType.EXACT_INPUT ? CurrencyAmount<TInput> : CurrencyAmount<TOutput>,
     tradeType: TTradeType
   ): Promise<MixedRouteTrade<TInput, TOutput, TTradeType>> {
-    const amounts: CurrencyAmount<Token>[] = new Array(route.path.length)
+    const amounts: CurrencyAmount<Currency>[] = new Array(route.path.length)
     let inputAmount: CurrencyAmount<TInput>
     let outputAmount: CurrencyAmount<TOutput>
 
     invariant(tradeType === TradeType.EXACT_INPUT, 'TRADE_TYPE')
-
     invariant(amount.currency.equals(route.input), 'INPUT')
-    amounts[0] = amount.wrapped
+
+    amounts[0] = route.pools[0] instanceof V4Pool ? amount : amount.wrapped
     for (let i = 0; i < route.path.length - 1; i++) {
       const pool = route.pools[i]
-      const [outputAmount] = await pool.getOutputAmount(amounts[i])
+      const [outputAmount] =
+        pool instanceof V4Pool ? await pool.getOutputAmount(amounts[i]) : await pool.getOutputAmount(amounts[i].wrapped)
```

Comment:
Ah it's going to be quite difficult always having to do the instanceof check - could we write a helper function that calls `getOutputAmount` with either the raw currency or the wrapped version depending on the instance of pool passed in?